src/Symfony/Component/HttpFoundation/File/MimeType/FileBinaryMimeTypeGuesser.php

```diff
@@ -82,4 +82,14 @@ public function guess($path)
 
         return $match[1];
     }
+
+    /**
+     * @return bool
+     */
+    private static function hasFileBinary()
+    {
+        static $exists;
+
+        return isset($exists) ? $exists : ($exists = null !== shell_exec('command -v file'));
```

Comment:
this should use `passthru` as well to run the command. Otherwise, it requires adding ` shell_exec` to the list of existing functions (as these are functions which are likely to get disabled on some hosting)